griptape/tasks/base_task.py

```diff
@@ -123,7 +122,7 @@ def execute(self) -> BaseArtifact:
             self.output = ErrorArtifact(str(e))
         finally:
             self.state = BaseTask.State.FINISHED
-            self.structure.publish_event(FinishTaskEvent(task=self))
+            self.structure.publish_event(FinishTaskEvent.from_task(self))
```

Comment:
Let's move it to `after_run`. And call it after line 116 (we are not currently calling it for some reason).

Reply:
Done. We are already calling `self.after_run()` on line 118, did you mean something else?

Reply:
Oops, nevermind—line 118 was folded in the PR code view.